improver/cli/normalise_to_reference.py

```diff
@@ -0,0 +1,106 @@
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""CLI to enforce the sum total of a set of forecasts to be equal to a reference."""
+
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(
+    *cubes: cli.inputcubelist,
+    reference_name: cli.comma_separated_list,
+    return_name: cli.comma_separated_list = None,
+    ignore_zero_total: bool = False,
+):
+    """Module to enforce that the sum of data in a list of cubes is equal to the
+    corresponding data in a reference cube.
+
+    The data are updated as follows, if there are 2 cubes to be normalised containing
+    data points a and b respectively with corresponding reference r, then:
+
+    a_new = r * (a / (a + b))
+    b_new = r * (b / (a + b))
+
+    which ensures that r = a_new + b_new.
+
+    Args:
+        cubes (List of iris.cube.Cube): A list of cubes containing both the cubes to be
+            updated and the reference cube. The reference cube will be identified by
+            matching the cube name to reference_name.
+        reference_name (str): The name of the reference cube, this should match exactly
+            one cube in cubes.
+        return_name (str): The name/s of the cubes to be returned. If None then all
+            updated cubes will be returned, otherwise only updated cubes with names
+            matching return_name will be returned. Defaults to None.
+        ignore_zero_total (bool): Determines whether an error will be raised in the
+            instance where the total of non-reference cubes is zero but the
+            corresponding value in reference is non-zero. If True this error will not be
+            raised - instead the data in the updated cubes will remain as zero values,
+            if False an error will be raised.
+
+    Returns:
+        iris.cube.Cube or iris.cube.CubeList:
+            A cubelist containing cubes with identical metadata to those in input cubes,
+            but with data updated so that the total of the data in these cubes is equal
+            to the reference. If only one cube is being returned, then a cube is
+            returned instead of a cubelist.
```

Comment:
As discussed, we will only support returning one Cube, not a list of Cubes, at least until we decide otherwise.

Reply:
Changed